### PR TITLE
Initialize trainer with None in DDPAccelerator

### LINKED ISSUE #4843: DDP accelorator should have 'trainer=none' as default
Add to DDP accelerator 'trainer=None' as a first argument (otherwise I cannot pass it to the Trainer instantiation)

https://github.com/PyTorchLightning/pytorch-lightning/blob/master/pytorch_lightning/accelerators/ddp_accelerator.py#L49

```
class DDPAccelerator(Accelerator):
    def __init__(self, trainer, cluster_environment=None, ddp_plugin=None)
```

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

Fixes #4843

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:  

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [ ] Check that all items from **Before submitting** are resolved
 - [ ] Make sure the title is self-explanatory and the description concisely explains the PR
 - [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified; _Bugfixes should be including in bug-fix release milestones (m.f.X) and features should be included in (m.X.b) releases._
 

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by edenlightning** [comment]
@tchaton sync?

**Review by SeanNaren — APPROVED** [review]
Nice, thanks for the cleanup

**Comment by carmocca** [comment]
Why arent all `trainer` arguments annotated as `Optional[Trainer]`?

**Comment by Borda** [comment]
> Why arent all `trainer` arguments annotated as `Optional[Trainer]`?

I think that belongs to #5023 :]

File: pytorch_lightning/accelerators/accelerator.py [hunk 1]
```diff
@@ -19,8 +19,12 @@
 import torch.distributed as torch_distrib
 from torch.optim import Optimizer
 
+import pytorch_lightning as pl
```
Comment (Borda) [inline pytorch_lightning/accelerators/accelerator.py:22]: pls do not use this call inside PL

File: pytorch_lightning/accelerators/accelerator.py [hunk 2]
```diff
@@ -33,7 +37,10 @@ class ReduceOp:
 
 class Accelerator(object):
 
-    def __init__(self, trainer=None, cluster_environment=None, ddp_plugin=None):
+    def __init__(self,
+                 trainer: Optional['pl.Trainer'] = None,
```
Comment (Borda) [inline pytorch_lightning/accelerators/accelerator.py:41]:
```suggestion
                 trainer: Optional['Trainer'] = None,
```
 as it is string you do not need the pl

**Review by Borda — APPROVED** [review]
not very sure in these shoes, but LGTM
just pls remove all the pl from imports and use the only trainer as typing

File: pytorch_lightning/accelerators/accelerator.py (hunk 1 above)
Comment (tchaton) [reply pytorch_lightning/accelerators/accelerator.py:22]: It doesn't work with the doc when I do.

File: pytorch_lightning/accelerators/accelerator.py (hunk 2 above)
Comment (tchaton) [reply pytorch_lightning/accelerators/accelerator.py:41]: You need it for the doc.

Comment (tchaton) [reply pytorch_lightning/accelerators/accelerator.py:41]: Same it breaks the doc.

**Comment by carmocca** [comment]
> > Why arent all `trainer` arguments annotated as `Optional[Trainer]`?
> 
> I think that belongs to #5023 :] but yes we shall not introduce more then really needed :]

`pl.Trainer` was added so might as well make it the correct type

**Comment by tchaton** [comment]
> > > Why arent all `trainer` arguments annotated as `Optional[Trainer]`?
> > 
> > 
> > I think that belongs to #5023 :] but yes we shall not introduce more then really needed :]
> 
> `pl.Trainer` was added so might as well make it the correct type

I removed it. I will try to sort out a better solution.

**Review by teddykoker — APPROVED** [review]
